frontend/src/scenes/feature-flags/FeatureFlag.tsx

```diff
@@ -520,225 +522,441 @@ function FeatureFlagRollout({ readOnly }: FeatureFlagReadOnlyProps): JSX.Element
+                                <Col span={4}>
+                                    Rollout
+                                    <LemonButton type="tertiary" onClick={distributeVariantsEqually}>
+                                        (Redistribute)
+                                    </LemonButton>
+                                </Col>
+                            </Row>
+                            {variants.map((variant, index) => (
+                                <Group key={index} name="filters">
+                                    <Row gutter={8} align="middle">
+                                        <Col span={1}>
+                                            <Lettermark name={alphabet[index]} color={LettermarkColor.Gray} />
+                                        </Col>
+                                        <Col span={4}>
+                                            <Field name={['multivariate', 'variants', index, 'key']}>
+                                                <LemonInput
+                                                    data-attr="feature-flag-variant-key"
+                                                    data-key-index={index.toString()}
+                                                    className="ph-ignore-input"
+                                                    placeholder={`example-variant-${index + 1}`}
+                                                    autoComplete="off"
+                                                    autoCapitalize="off"
+                                                    autoCorrect="off"
+                                                    spellCheck={false}
+                                                />
+                                            </Field>
+                                        </Col>
+                                        <Col span={6}>
+                                            <Field name={['multivariate', 'variants', index, 'name']}>
+                                                <LemonInput
+                                                    data-attr="feature-flag-variant-name"
+                                                    className="ph-ignore-input"
+                                                    placeholder="Description"
+                                                />
+                                            </Field>
+                                        </Col>
+                                        <Col span={8}>
+                                            <Field name={['payloads', index]}>
+                                                {({ value, onChange }) => {
+                                                    const valToShow =
+                                                        index in featureFlag.filters.payloads
+                                                            ? value
+                                                            : featureFlag.filters.payloads?.[variant.key]
```

Comment:
This is causing some strange behavior here 

https://user-images.githubusercontent.com/25164963/212203589-aa703bff-89bf-4933-8797-f26a854786ff.mov

Reply:
Yeah, been thinking on this. Need to rework the form logic to get rid of this nasty mapping

Reply:
One good way around this (which I found thanks to the hours spent on the experiment UI) is to directly control the payload input here

```
<JSONEditorInput onChange={(newVal) => {setFeatureFlagValue('filters', {...existingFilters, 'payloads': {...existingPayloads, 'variant': newVal}}} value={valToShow} />
```

The `set*Value` is an action automatically generated from kea-forms, where * = the form in question.

Reply:
Ideally this would be wrapped in another element so it can be re-used without issues with both non-boolean and multivariate flags, based on the 'variant' input.

Reply:
Solution atm is that variant payloads are stored by index and then mapped to/from retrieval from backend. So when saving, indexs -> keys and when retrieved keys->indexes. This seems to be a reasonable way to track this form because you can end up with states where the variant object has no key so the key is not completely reliable to store against but the index is